new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioCrossDomainTransaction.java

```diff
@@ -0,0 +1,426 @@
+    dockerLoginAndPushImageToRegistry(domain1Image);
+
+    logger.info("Creating image with model file and verify");
+    String domain2Image = createImageAndVerify(
+        WDT_IMAGE_NAME2, WDT_MODEL_FILE_DOMAIN2, appSource, WDT_MODEL_DOMAIN2_PROPS, PROPS_TEMP_DIR, domainUid2);
+    logger.info("Created {0} image", domain2Image);
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(domain2Image);
+
+    //create domain1
+    createDomain(domainUid1, domain1Namespace, domain1AdminSecretName, domain1Image);
+    //create domain2
+    createDomain(domainUid2, domain2Namespace, domain2AdminSecretName, domain2Image);
+
+    String clusterService = domainUid1 + "-cluster-" + clusterName + "." + domain1Namespace + ".svc.cluster.local";
+
+    Map<String, String> templateMap  = new HashMap();
+    templateMap.put("NAMESPACE", domain1Namespace);
+    templateMap.put("ADMIN_SERVICE",domain1AdminServerPodName);
+    templateMap.put("CLUSTER_SERVICE", clusterService);
+
+    Path svcYamlSrc = Paths.get(RESOURCE_DIR, "istio", "istio-cdt-http-template-service.yaml");
+    Path svcYmlTarget = assertDoesNotThrow(
+        () -> generateFileFromTemplate(svcYamlSrc.toString(),
+            "istiocrossdomaintransactiontemp/istio-cdt-http-service.yaml", templateMap));
+    logger.info("Generated Http VS/Gateway file path is {0}", svcYmlTarget);
+
+    boolean deployRes = deployHttpIstioGatewayAndVirtualservice(svcYmlTarget);
+    assertTrue(deployRes, "Could not deploy Http Istio Gateway/VirtualService");
+
+    int istioIngressPort = getIstioHttpIngressPort();
+    logger.info("Istio Ingress Port is {0}", istioIngressPort);
+
+    logger.info("Validating WebLogic admin server access by login to console");
+    boolean loginSuccessful = assertDoesNotThrow(() -> {
+      return adminNodePortAccessible(istioIngressPort, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+    }, "Access to admin server node port failed");
+    assertTrue(loginSuccessful, "Console login validation failed");
+
+
+    String curlRequest = String.format("curl -v --show-error --noproxy '*' "
+            + "http://%s:%s/TxForward/TxForward?urls=t3://%s.%s:7001,t3://%s1.%s:8001,t3://%s1.%s:8001,"
+            + "t3://%s2.%s:8001",
+             K8S_NODEPORT_HOST, istioIngressPort, domain1AdminServerPodName, domain1Namespace,
+             domain1ManagedServerPrefix, domain1Namespace, domain2ManagedServerPrefix,domain2Namespace,
+             domain2ManagedServerPrefix,domain2Namespace);
+
+    ExecResult result = null;
+    logger.info("curl command {0}", curlRequest);
+    result = assertDoesNotThrow(
+        () -> exec(curlRequest, true));
+    if (result.exitValue() == 0) {
+      logger.info("\n HTTP response is \n " + result.stdout());
+      logger.info("curl command returned {0}", result.toString());
+      assertTrue(result.stdout().contains("Status=Committed"), "crossDomainTransaction failed");
+    }
```

Comment:
Do you want to use java based http client?

String url= String.format("
            "http://%s:%s/TxForward/TxForward?urls=t3://%s.%s:7001,t3://%s1.%s:8001,t3://%s1.%s:8001,"
            + "t3://%s2.%s:8001",
             K8S_NODEPORT_HOST, istioIngressPort, domain1AdminServerPodName, domain1Namespace,
             domain1ManagedServerPrefix, domain1Namespace, domain2ManagedServerPrefix,domain2Namespace,
             domain2ManagedServerPrefix,domain2Namespace);

HttpResponse<String> response = assertDoesNotThrow(() -> OracleHttpClient.get(url, true));
assertEquals(200, response.statusCode(), "Status code not equals to 200");
assertTrue(response.body().contains(""Status=Committed"), "crossDomainTransaction failed");

Reply:
changed to use the http client